packages/core/versioning/core-version-http-browser/src/version_http_toolkit.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { HttpSetup } from '@kbn/core-http-browser';
+
+/**
+ * A number > 0.
+ * @experimental
+ */
+export type Version = `${number}`;
+
+/** @experimental */
+export interface CreateClientArgs {
+  /** @experimental */
+  version: Version;
+}
+
+/** @experimental */
+export interface VersionHTTPToolkit {
+  /**
+   * Create a new HTTP client for a specific version.
```

Comment:
Having to create a new client for every version seems like a lot of overhead. I'd guess that ideally, we'd use the same base HTTP client with an option to specify the version. I'd have to think about the implementation differences though.